doc/03_reference/testplan.rst

```diff
@@ -0,0 +1,109 @@
+  One monitors the RVFI interface which provides details of retired instructions.
+  The other monitors relate to fetched instructions and instruction memory errors, more details are provided in :ref:`cosim`.
+  Additionally it connects to the monitor of the Memory Interface Agent for the instruction and data side via analysis ports.
+  The monitored transactions are used by a scoreboard to provide information to the co-simulation system allowing it to step the golden model and check its execution and memory activity against Ibex's behaviour.
+
+Memory Interface Agent
+  This provides a driver and a monitor for the :ref:`Ibex Memory Interface Protocol<lsu-protocol>`.
+  The driver provides fully randomised and configurable timings for responses and randomisation of error responses.
+  Two agents are instaniated, one for the data memory interface the other for the instruction memory interface.
+  Read data for memory responses is provided from a backing memory, writes requests update the contents of the backing memory.
+  This is separate from the memory used by the golden model in the co-simulation agent.
+  The contents of these two memories will be identical unless there is a mismatch resulting in a failure.
+  The backing memory is held in a memory model as a separate UVM component.
+  The two agents use the same backing memory so they have a coherent view of memory.
+
+IRQ Agent
+  This provides a driver and a monitor for the IRQ interface.
+  It provides randomised interrupt stimulus to Ibex when a test requests it.
+  Constraints can be used to control types of interrupts generated (e.g. NMI or not) and whether multiple interrupts should be raised together.
+
+Debug and reset signals are a single wire each so do not have a dedicated agent.
+Instead any sequence that wishes to use them will directly manipulate them via a virtual interface
+
+The testbench instantiates the agents described above along with the memory model used by both the data and instruction side memory agents.
+A test consists of executing a pre-built binary (which is loaded into the memory model at the start of the test via backdoor accesses) along with configuring agents to provide appropriate stimulus for the test.
+Some tests may use the agents to generate stimulus at particular times (e.g.  interrupts).
+A test may perform additional checking on top of the co-simulation golden model comparison where appropriate (e.g. ensuring a raised interrupt has cause an exception).
+
+Stimulus Strategy
+-----------------
+
+Stimulus falls into two categories:
+
+* Instructions to execute, these are generated by the `RISC-V DV random instruction <https://github.com/google/riscv-dv>`_ generator and provided to the testbench via a raw binary file.
+* Activity on external interfaces.
+
+Instructions are generated ahead of the test run so the test has no control over them.
+All external interfaces have their stimulus generated during the test run so can be controlled by the test.
```

Comment:
The combination of "test run" (the execution itself) and "test" (some form of test specification) may be confusing here. Also, the test actually specifies which instructions will be tested (e.g. bitmanip yes/no, wfi etc.), i.e., the test has an influence on the instructions, it's just offline.

Would it make sense to rephrase this as follows?
"Instructions are generated ahead of time so the test has no control over them at run time."
"All external interfaces have their stimulus generated at run time so can be controlled by the test."